apps/emqx_resource/src/emqx_resource_manager.erl

```diff
@@ -649,10 +656,8 @@ do_wait_for_ready(ResId, Retry) ->
 
 safe_call(ResId, Message, Timeout) ->
     try
-        Module = atom_to_binary(?MODULE),
-        MgrId = get_owner(ResId),
-        ProcName = binary_to_existing_atom(<<Module/binary, "_", MgrId/binary>>, utf8),
-        gen_statem:call(ProcName, Message, {clean_timeout, Timeout})
+        ManagerPid = read_manager_pid_from_cache(ResId),
+        gen_statem:call(ManagerPid, Message, {clean_timeout, Timeout})
```

Comment:
Since we now rely on the table, maybe we could just match on the lookup result rather than raising errors.  Ex:

```suggestion
        case read_cache(ResId) of
            not_found ->
                {error, not_found};
            ManagerPid ->
                gen_statem:call(ManagerPid, Message, {clean_timeout, Timeout})
        end
```